Shared/Samples/Run valve isolation trace/RunValveIsolationTraceView.swift

```diff
@@ -0,0 +1,182 @@
+            }
+            .task(id: lastSingleTap?.mapPoint) {
+                guard let lastSingleTap = lastSingleTap else {
+                    return
+                }
+                if let feature = try? await mapViewProxy.identifyLayers(
+                    screenPoint: lastSingleTap.screenPoint,
+                    tolerance: 10
+                ).first?.geoElements.first as? ArcGISFeature {
+                    model.addFilterBarrier(for: feature, at: lastSingleTap.mapPoint)
+                }
+            }
+            .toolbar {
+                ToolbarItemGroup(placement: .bottomBar) {
+                    Button {
+                        isConfigurationPresented.toggle()
+                    } label: {
+                        Text("Configuration")
+                    }
+                    .disabled(model.tracingActivity == .runningTrace)
+                    Spacer()
+                    Button("Trace") {
+                        Task { await model.trace() }
+                    }
+                    .disabled(!model.traceEnabled)
+                    Spacer()
+                    Button("Reset") {
+                        model.reset()
+                        Task { await mapViewProxy.setViewpointCenter(model.startingLocationPoint, scale: 3_000) }
+                    }
+                    .disabled(!model.resetEnabled)
+                }
+            }
+            .sheet(isPresented: $isConfigurationPresented) {
+                if #available(iOS 16, *) {
+                    NavigationStack {
+                        configurationView
+                    }
+                } else {
+                    NavigationView {
+                        configurationView
+                    }
+                }
+            }
+            .overlay(alignment: .center) {
+                if model.tracingActivity != nil {
+                    VStack {
+                        Text(model.tracingActivity?.label ?? "")
+                        ProgressView()
+                            .progressViewStyle(.circular)
+                    }
+                    .padding(6)
+                    .background(.thinMaterial)
+                    .cornerRadius(10)
+                }
+            }
+            .alert(
+                "Select terminal",
+                isPresented: $model.terminalSelectorIsOpen,
+                actions: { terminalPickerButtons }
+            )
+        }
+    }
+    
+    /// Buttons for each the available terminals on the last added utility element.
+    @ViewBuilder var terminalPickerButtons: some View {
+        ForEach(model.lastAddedElement?.assetType.terminalConfiguration?.terminals ?? []) { terminal in
+            Button(terminal.name) {
+                model.lastAddedElement?.terminal = terminal
+                model.terminalSelectorIsOpen = false
+                model.addTerminal(to: lastSingleTap!.mapPoint)
+            }
+        }
```

Comment:
Some things are being unwrapped multiple times here. Better to unwrap them up front, improving readability.

```suggestion
        if let lastAddedElement = model.lastAddedElement,
           let terminalConfiguration = lastAddedElement.assetType.terminalConfiguration {
            ForEach(terminalConfiguration.terminals) { terminal in
                Button(terminal.name) {
                    lastAddedElement.terminal = terminal
                    model.terminalSelectorIsOpen = false
                    model.addTerminal(to: lastSingleTap!.mapPoint)
                }
            }
        }
```